test/integration/framework/test_server.go

```diff
@@ -147,9 +168,12 @@ func StartTestServer(t *testing.T, stopCh <-chan struct{}, setup TestServerSetup
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	errCh = make(chan error)
 	go func() {
+		defer close(errCh)
 		if err := kubeAPIServer.GenericAPIServer.PrepareRun().Run(stopCh); err != nil {
-			t.Error(err)
+			errCh <- err
```

Comment:
same comment as in https://github.com/kubernetes/kubernetes/pull/110000/files#r882448491, this channel is only check on cleanup, we can do a quick check when waiting for healthz to fail fast instead of waiting for the apiserver /healthz endpoint

Reply:
good point - added